packages/app/src/api.ts

```diff
@@ -171,6 +179,114 @@ export function initializeApp(
   return newApp;
 }
 
+/**
+ * Creates and initializes a {@link @firebase/app#FirebaseServerApp} instance.
+ *
+ * The FirebaseServerApp is similar to FirebaseApp, but is intended for execution in
+ * server side rendering environments only.
+ *
+ * See
+ * {@link
+ *   https://firebase.google.com/docs/web/setup#add_firebase_to_your_app
+ *   | Add Firebase to your app} and
+ * {@link
+ *   https://firebase.google.com/docs/web/setup#multiple-projects
+ *   | Initialize multiple projects} for detailed documentation.
+ *
+ * @example
+ * ```javascript
+ *
+ * // Initialize a FirebaseServerApp.
+ * // Retrieve your own options values by adding a web app on
+ * // https://console.firebase.google.com
+ * initializeServerApp({
+ *     apiKey: "AIza....",                             // Auth / General Use
+ *     authDomain: "YOUR_APP.firebaseapp.com",         // Auth with popup/redirect
+ *     databaseURL: "https://YOUR_APP.firebaseio.com", // Realtime Database
+ *     storageBucket: "YOUR_APP.appspot.com",          // Storage
+ *     messagingSenderId: "123456789"                  // Cloud Messaging
+ *   },
+ *   {
+ *    authIdToken: "Your Auth ID Token"
+ *   });
+ * ```
+ *
+ * @param options - Firebase.AppOptions to configure the app's services, or a
+ *   a FirebaseApp instance which contains the AppOptions within.
+ * @param config - FirebaseServerApp configuration.
+ *
+ * @returns The initialized FirebaseServerApp.
+ *
+ * @public
+ */
+export function initializeServerApp(
+  options: FirebaseOptions | FirebaseApp,
+  config: FirebaseServerAppSettings
+): FirebaseServerApp;
+
+export function initializeServerApp(
+  _options: FirebaseOptions | FirebaseApp,
+  _serverAppConfig: FirebaseServerAppSettings
+): FirebaseServerApp {
```

Comment:
we should throw if they're in a client side browser env. easiest test would be `typeof window` but we might want to be smarter

Reply:
Done!